nexus/db-model/src/resource_lock.rs

```diff
@@ -0,0 +1,14 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::schema::resource_lock;
+use uuid::Uuid;
+
+/// Database representation of a resource lock
+#[derive(Queryable, Insertable, Debug, Clone, Selectable)]
+#[diesel(table_name = resource_lock)]
+pub struct ResourceLock {
+    pub resource_id: Uuid,
```

Comment:
Might be worth documenting:

`resource_id` is the object being locked

I actually think `lock_id` is acting more like `owner_id` -- it should uniquely identify "who is asking for the lock", rather than the lock itself.

For example, in a theoretical situation where we grab multiple locks (... and, on that note, we should consider that deadlocks are possible now), it would be "normal" to use the same `lock_id` value across multiple `ResourceLock` objects.